### PR TITLE
[bugfix] Predicate statistics calculation considers the evaluation of EQ_FOR_NULL expressions

### PR DESCRIPTION
**Pull request by HangyuanLiu**
## What type of PR is this：
- [x] bugfix
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

At present, we do not distinguish the difference between EQ and EQ_FOR_NULL when calculating the predicate filtering degree. This results in incorrect estimates for scenarios such as col <=> null. The modification of this PR handles this scenario separately, while in other scenarios, the processing logic remains the same as before.
## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 23 / 27 (85.19%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/statistic/StatisticUtils.java|8|12|66.67%|[217, 231, 232, 233]|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/BinaryPredicateStatisticCalculator.java|15|15|100.00%|[]|

**Review by kangkaisen — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/statistic/StatisticUtils.java [hunk 1]
```diff
@@ -204,24 +208,29 @@ public static Optional<Double> convertStatisticsToDouble(Type type, String stati
...
+                case DATE:
+                    return Optional.of((double) getLongFromDateTime(DateUtils.parseStringWithDefaultHSM(
+                            statistic, DateUtils.DATE_FORMATTER_UNIX)));
+                case DATETIME:
+                    return Optional.of((double) getLongFromDateTime(DateUtils.parseStringWithDefaultHSM(
+                            statistic, DateUtils.DATE_TIME_FORMATTER_UNIX)));
+                case CHAR:
+                case VARCHAR:
+                    return Optional.empty();
+                default:
```
Comment (packy92) [inline fe/fe-core/src/main/java/com/starrocks/statistic/StatisticUtils.java:228]: Can we add a branch to process NULL type here?

Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/statistic/StatisticUtils.java:228]: No, NULL values should not enter this function, and our current implementation logic is that NULL Literal can be of any type, and the type depends on the calculation involved

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 23 / 28 (82.14%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/statistic/StatisticUtils.java|8|13|61.54%|[217, 231, 232, 233, 234]|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/BinaryPredicateStatisticCalculator.java|15|15|100.00%|[]|

**Review by Youngwb — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**